website/files/models/osfstorage.py

```diff
@@ -82,9 +83,39 @@ def move_under(self, destination_parent, name=None):
         return super(OsfStorageFileNode, self).move_under(destination_parent, name)
 
     def save(self):
+        """
+        Saves node, logs any check-out or check-in actions
+            but we want to log specific activities (checkout, checkin)
+            check that via the if in the middle of the method and log the correct activity
+
+        :return:
+        """
         self.path = ''
         self.materialized_path = ''
-        return super(OsfStorageFileNode, self).save()
+        ret = super(OsfStorageFileNode, self).save()
+
+        if 'checkout' in ret and len(ret) == 1:
```

Comment:
I'm not sure about this approach. Typically, we would add a model method that would make field changes and do logging.

```python
file_node.check_out(...)  # sets 'checkout' and adds node log
```

I think this would be cleaner than adding all this logic to `save()`.